File: LeetCode/removeElement.java
//https://leetcode.com/problems/remove-element/
//Title: Remove Element
//Difficulty: Easy
//Language: Java
//Author: Pranay Kotian

class Solution {
    public int removeElement(int[] nums, int val) {
        /*
        int length = 0;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] == val) {
                i -= 1;
                for (int j = i + 1; j < nums.length; j++) {
                    nums[j - 1] = nums[j];
                }
            }
            else
                length += 1;
        }
        return length + 1; 

        //Solution exceeds time limit when i is decremented by 1
        */
        
        /*
        int i = 0;
        int len = nums.length; 
        while (i < len) {
            if (nums[i] == val) {
                for (int j = i + 1; j < len; j++)
                    nums[j - 1] = nums[j];
            }
            else
                i += 1; 
        }
        return i + 1;

        //Solution exceeds tie limit
        */
    }
}

File: LeetCode/removeElement.py
#https://leetcode.com/problems/remove-element/
#Title: Remove Element
#Difficulty: Easy
#Language: Python3
#Author: Pranay Kotian

class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        #Identical solution to remove duplicates problem
        i = 0 
        while (i < len(nums)):
            if (nums[i] == val):
                nums.remove(val)
            else:
                i += 1 
        return i + 1

        """
        Runtime: 28 ms, faster than 84.57% of Python3 online submissions for Remove Element.
		Memory Usage: 12.7 MB, less than 100.00% of Python3 online submissions for Remove Element.
        """